ple/games/vgdlple.py

```diff
@@ -0,0 +1,168 @@
+import sys
+import pygame
+import base
+import vgdl.core as core
+
+# Hacked together VLE wrapper for py-VGDL. Potentially extremely buggy - use with caution.
+# This allows you to use VGDL (see http://people.idsia.ch/~tom/publications/pyvgdl.pdf , thanks to Tom Schaul) with the PLE.
+
+# Example VGDL description text
+# The game dynamics are specified as a paragraph of text
+aliens_game = """
+BasicGame
+    SpriteSet
+        base    > Immovable    color=WHITE
+        avatar  > FlakAvatar   stype=sam
+        missile > Missile
+            sam  > orientation=UP    color=BLUE singleton=True
+            bomb > orientation=DOWN  color=RED  speed=0.5
+        alien   > Bomber       stype=bomb   prob=0.01  cooldown=3 speed=0.75
+        portal  > SpawnPoint   stype=alien  cooldown=16   total=20
+    
+    LevelMapping
+        0 > base
+        1 > portal
+
+    TerminationSet
+        SpriteCounter      stype=avatar               limit=0 win=False
+        MultiSpriteCounter stype1=portal stype2=alien limit=0 win=True
+        
+    InteractionSet
+        avatar  EOS  > stepBack
+        alien   EOS  > turnAround        
+        missile EOS  > killSprite
+        missile base > killSprite
+        base missile > killSprite
+        base   alien > killSprite
+        avatar alien > killSprite
+        avatar bomb  > killSprite
+        alien  sam   > killSprite         
+"""
+
+# the (initial) level as a block of characters 
+aliens_level = """
+wwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww
+w                              w
+w1                             w
+w000                           w
+w000                           w
+w                              w
+w                              w
+w                              w
+w                              w
+w    000      000000     000   w
+w   00000    00000000   00000  w
+w   0   0    00    00   00000  w
+w                A             w
+wwwwwwwwwwwwwwwwwwwwwwwwwwwwwwww
+"""
+
+
+class vgdlPLE(base.PyGameWrapper):
+
+
+    def __init__(self, game_str = aliens_game, map_str = aliens_level):
+
+	# GVG-AI competition allows 40ms per move, which equates to 25fps
+        self.allowed_fps = 25
+
+        self.game_desc = game_str
+        self.level_desc = map_str
+        
+        # Need to build a sample level to get the available actions and screensize....
+        self.g = core.VGDLParser().parseGame(self.game_desc)
+        self.g.buildLevel(self.level_desc)
+
+        actions = self.g.getPossibleActions()
+        base.PyGameWrapper.__init__(self, self.g.screensize[0], self.g.screensize[1], actions=actions)
+
+
+    def getGameState(self):
+        # Using the VGDL State Representation for now
+        return self.g.getFullState()
+
+    def getScore(self):
+        return self.g.score
+
+    def game_over(self):
+        return self.g.ended
+
+    def init(self):
```

Comment:
general question: if the game must restart often, as would happen if the player losses immediately over and over at the start, how fast is the init method? I ran into an issue with this on some games init methods and had to do some 'caching' to speed it up. It seems the construction of each game is fairly involved. Is there an intermediate state or even an initial clean copy we can keep to reset from?